kodilla-hibernate/src/main/java/com/kodilla/hibernate/manytomany/facade/Facade.java

```diff
@@ -0,0 +1,27 @@
+package com.kodilla.hibernate.manytomany.facade;
+
+import com.kodilla.hibernate.manytomany.Company;
+import com.kodilla.hibernate.manytomany.Employee;
+import com.kodilla.hibernate.manytomany.dao.CompanyDao;
+import com.kodilla.hibernate.manytomany.dao.EmployeeDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class Facade {
+
+    @Autowired
+    CompanyDao companyDao;
+    @Autowired
+    EmployeeDao employeeDao;
+
+    public List<Company> findCompaniesByNameFragment(String arg) {
+        return companyDao.findCompaniesByNameFragment("%" + arg + "%");
+    }
+
+    public List<Employee> findEmployeesByNameFragment(String arg) {
+        return employeeDao.findEmployeesByNameFragment("%" + arg + "%");
+    }
```

Comment:
Moim zdaniem to rozwiązanie nie różni się niczym szczególnym w porównaniu z poprzednią wersją tego zadania. Ja bym wystawil w fasadzie jedną metodę `findByName`, która przyjmnie za argument jakiś fragment nazwy, zaś zwróci wszystkich pracowników, których nazwiska lub imiona zawierają ten fragment oraz wszystkie firmy, które w nazwie zawierają ten fragment. Do tego wprowadzić trzebaby dodatkowy obiekt, nazwijmy go roboczo `FindByNameResponse` - będzie miał dwa pola: listę pracowników oraz listę firm. Chodzi o to, aby fasada miała jak najmniej metod: cala logika wewnątrz. U ciebie fasada jest totalnie niepotrzebna - zawiera tylko przelotki do dwóch dao. :)